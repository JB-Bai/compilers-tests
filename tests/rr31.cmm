float ghDjTFqYj(struct Vu19guRQNdB{
	int x52X7ogkd[90];
}
YH6h8IrB, int rvd7Q3[9540][7790475], float QzD0Txr60, struct F72H CSSiSnU, int tzfX4I2jYIe[0136], struct Ifm1F7 cFRYobpJh[0x528][150], struct _0MopOE{
}
mo4IVk5V[0x67c]){
	return 3873;
	42;
	if(qbkpX) CUWf5KKh * 16.189E7 - yjymih(lEjMd5tf(Vqm4Z, JnVJ3b1nmD) && BNQ4WL1,(YlV_uH == 0x2b5 && jVwzulpI || fIR2RlHlSg(0x4) && yS2ZVr) >= jkYPsDXJ, 33.511.YZCiAxQeX + 0727);
	047757 < 9.626 - r10q5X3qFw() = 8.08 && 0x086 / tdtWC4 * 461 = 20905 - 8.508E+9;
}
struct fZjMdRRB;
float;

