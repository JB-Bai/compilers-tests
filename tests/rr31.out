Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      FunDec (1)
        ID: ghDjTFqYj
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              StructSpecifier (1)
                STRUCT
                OptTag (1)
                  ID: Vu19guRQNdB
                LC
                DefList (2)
                  Def (2)
                    Specifier (2)
                      TYPE: int
                    DecList (2)
                      Dec (2)
                        VarDec (2)
                          VarDec (2)
                            ID: x52X7ogkd
                          LB
                          INT: 90
                          RB
                    SEMI
                RC
            VarDec (4)
              ID: YH6h8IrB
          COMMA
          VarList (4)
            ParamDec (4)
              Specifier (4)
                TYPE: int
              VarDec (4)
                VarDec (4)
                  VarDec (4)
                    ID: rvd7Q3
                  LB
                  INT: 9540
                  RB
                LB
                INT: 7790475
                RB
            COMMA
            VarList (4)
              ParamDec (4)
                Specifier (4)
                  TYPE: float
                VarDec (4)
                  ID: QzD0Txr60
              COMMA
              VarList (4)
                ParamDec (4)
                  Specifier (4)
                    StructSpecifier (4)
                      STRUCT
                      Tag (4)
                        ID: F72H
                  VarDec (4)
                    ID: CSSiSnU
                COMMA
                VarList (4)
                  ParamDec (4)
                    Specifier (4)
                      TYPE: int
                    VarDec (4)
                      VarDec (4)
                        ID: tzfX4I2jYIe
                      LB
                      INT: 94
                      RB
                  COMMA
                  VarList (4)
                    ParamDec (4)
                      Specifier (4)
                        StructSpecifier (4)
                          STRUCT
                          Tag (4)
                            ID: Ifm1F7
                      VarDec (4)
                        VarDec (4)
                          VarDec (4)
                            ID: cFRYobpJh
                          LB
                          INT: 1320
                          RB
                        LB
                        INT: 150
                        RB
                    COMMA
                    VarList (4)
                      ParamDec (4)
                        Specifier (4)
                          StructSpecifier (4)
                            STRUCT
                            OptTag (4)
                              ID: _0MopOE
                            LC
                            RC
                        VarDec (6)
                          VarDec (6)
                            ID: mo4IVk5V
                          LB
                          INT: 1660
                          RB
        RP
      CompSt (6)
        LC
        StmtList (7)
          Stmt (7)
            RETURN
            Exp (7)
              INT: 3873
            SEMI
          StmtList (8)
            Stmt (8)
              Exp (8)
                INT: 42
              SEMI
            StmtList (9)
              Stmt (9)
                IF
                LP
                Exp (9)
                  ID: qbkpX
                RP
                Stmt (9)
                  Exp (9)
                    Exp (9)
                      Exp (9)
                        ID: CUWf5KKh
                      STAR
                      Exp (9)
                        FLOAT: 161890000.000000
                    MINUS
                    Exp (9)
                      ID: yjymih
                      LP
                      Args (9)
                        Exp (9)
                          Exp (9)
                            ID: lEjMd5tf
                            LP
                            Args (9)
                              Exp (9)
                                ID: Vqm4Z
                              COMMA
                              Args (9)
                                Exp (9)
                                  ID: JnVJ3b1nmD
                            RP
                          AND
                          Exp (9)
                            ID: BNQ4WL1
                        COMMA
                        Args (9)
                          Exp (9)
                            Exp (9)
                              LP
                              Exp (9)
                                Exp (9)
                                  Exp (9)
                                    Exp (9)
                                      ID: YlV_uH
                                    RELOP
                                    Exp (9)
                                      INT: 693
                                  AND
                                  Exp (9)
                                    ID: jVwzulpI
                                OR
                                Exp (9)
                                  Exp (9)
                                    ID: fIR2RlHlSg
                                    LP
                                    Args (9)
                                      Exp (9)
                                        INT: 4
                                    RP
                                  AND
                                  Exp (9)
                                    ID: yS2ZVr
                              RP
                            RELOP
                            Exp (9)
                              ID: jkYPsDXJ
                          COMMA
                          Args (9)
                            Exp (9)
                              Exp (9)
                                Exp (9)
                                  FLOAT: 33.511002
                                DOT
                                ID: YZCiAxQeX
                              PLUS
                              Exp (9)
                                INT: 471
                      RP
                  SEMI
              StmtList (10)
                Stmt (10)
                  Exp (10)
                    Exp (10)
                      Exp (10)
                        INT: 20463
                      RELOP
                      Exp (10)
                        Exp (10)
                          FLOAT: 9.626000
                        MINUS
                        Exp (10)
                          ID: r10q5X3qFw
                          LP
                          RP
                    ASSIGNOP
                    Exp (10)
                      Exp (10)
                        Exp (10)
                          FLOAT: 8.080000
                        AND
                        Exp (10)
                          Exp (10)
                            Exp (10)
                              INT: 134
                            DIV
                            Exp (10)
                              ID: tdtWC4
                          STAR
                          Exp (10)
                            INT: 461
                      ASSIGNOP
                      Exp (10)
                        Exp (10)
                          INT: 20905
                        MINUS
                        Exp (10)
                          FLOAT: 8508000256.000000
                  SEMI
        RC
    ExtDefList (12)
      ExtDef (12)
        Specifier (12)
          StructSpecifier (12)
            STRUCT
            Tag (12)
              ID: fZjMdRRB
        SEMI
      ExtDefList (13)
        ExtDef (13)
          Specifier (13)
            TYPE: float
          SEMI
